### ISSUE TITLE
Carrier-grade NAT may affect websocket connection

### ISSUE THREAD

**Issue by danielsz** [taoensso/sente#441]
[Carrier-grade NAT](https://en.wikipedia.org/wiki/Carrier-grade_NAT) is a practice by ISPs that allows them to share small pools of public addresses among many end users. I believe that in certain circumstances, carrier-grade NAT may affect the stability of webcocket connections, possibly in non-trivial developer setups or otherwise. Some time ago, I was observing constant interruptions of websocket connections. I called my ISP and they told me that they were aware of the problem, explained it was due to carrier-grade NAT, and upgraded my subscription. No more carrier-grade NAT, now I get assigned a public IP address that is not shared (still dynamic though), and that was the end the interruptions with websocket connections.
Unfortunately, I am fuzzy about the details since that was some time ago.

**Comment by ptaoussanis** [comment]
@danielsz Thanks Daniel, much appreciated! 🙏

I'll do some digging into this the next time I'm on batched Sente work.

In the meantime, I'd like to please ask anyone else that may have experienced something similar to 👍 and/or share extra details if they have any. Thanks!

**Comment by ptaoussanis** [comment]
@danielsz One question so long: when you observed the flaky WebSocket behaviour - was it actually in the context of Sente, or elsewhere?

**Comment by danielsz** [comment]
Yes, in the context of Sente, developing a project. The crucial part I don't remember is if it was all websocket connections or just my developer environment which was behind a local tunnel.

**Comment by ptaoussanis** [comment]
For reference, there was another [report](https://github.com/taoensso/sente/issues/435) recently that [suggested](https://github.com/taoensso/sente/issues/435#issuecomment-1743820570) an ISP doing "traffic filtering". Unfortunately didn't get any concrete details, but mentioning here in case it's related.

**Comment by ptaoussanis** [comment]
Closing as part of issue triage since there's been no further updates, reports, or 👍s in a few years. But please feel free to reopen if this is still relevant!
